File: app/src/main/java/com/disruption/cookcentral/ui/favs/FavouritesViewModelFactory.java
package com.disruption.cookcentral.ui.favs;

import android.app.Application;

import androidx.annotation.NonNull;
import androidx.lifecycle.ViewModel;
import androidx.lifecycle.ViewModelProvider;

@SuppressWarnings("ALL")
public class FavouritesViewModelFactory extends ViewModelProvider.NewInstanceFactory {
    private Application mApplication;

    public FavouritesViewModelFactory(Application application) {
        mApplication = application;
    }

    @NonNull
    @Override
    public <T extends ViewModel> T create(@NonNull Class<T> modelClass) {
        if (modelClass.isAssignableFrom(FavouritesViewModel.class)) {
            return (T) new FavouritesViewModel(mApplication);
        }

        throw new IllegalArgumentException("Unknown ViewModel class");
    }
}

File: app/src/main/java/com/disruption/cookcentral/ui/favs/FavouritesViewModelFactory.kt
package com.disruption.cookcentral.ui.favs

import android.app.Application
import androidx.lifecycle.ViewModel
import androidx.lifecycle.ViewModelProvider.NewInstanceFactory

@Suppress("UNCHECKED_CAST")
class FavouritesViewModelFactory(private val mApplication: Application) : NewInstanceFactory() {
    override fun <T : ViewModel?> create(modelClass: Class<T>): T {
        if (modelClass.isAssignableFrom(FavouritesViewModel::class.java)) {
            return FavouritesViewModel(mApplication) as T
        }
        throw IllegalArgumentException("Unknown ViewModel class")
    }

}